solutions/instances/main.tf

```diff
@@ -49,6 +50,8 @@ locals {
       can(regex(".*hs-crypto.*", var.existing_kms_crn)) ? "hs-crypto" : null
     )
   ) : null
+
+  kms_region = element(split(":", var.existing_kms_crn), length(split(":", var.existing_kms_crn)) - 5)
```

Comment:
can we rename `existing_kms_crn` to `existing_kms_instance_crn` so it is clear its the instance CRN here.

Also `existing_kms_crn` is optional as it is not needed if user is passing in a value for `existing_cos_kms_key_crn` OR if they are passing existing buckets. So this logic needs to be more advanced...
- If not passing in existing buckets, parse KMS region from `existing_cos_kms_key_crn`, otherwise get it from `existing_kms_instance_crn`. 

I notice the variable description for `existing_kms_crn` says "Only required if not supplying an existing KMS root key and if 'skip_cos_kms_auth_policy' is true." however you need to remove "and if 'skip_cos_kms_auth_policy' is true." from that description and also mention its not needed if passing values for existing buckets.